custom_components/frigate/update.py

```diff
@@ -0,0 +1,102 @@
+"""Update platform for frigate."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.update import UpdateEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_MODEL, CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import FrigateDataUpdateCoordinator, get_frigate_device_identifier
+from .const import ATTR_COORDINATOR, DOMAIN, NAME
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Sensor entry setup."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][ATTR_COORDINATOR]
+
+    entities = []
+    entities.append(FrigateContainerUpdate(coordinator, entry))
+    async_add_entities(entities)
+
+
+class FrigateContainerUpdate(UpdateEntity, CoordinatorEntity):  # type: ignore[misc]
+    """Frigate container update."""
+
+    _attr_title = "Frigate Container"
+
+    def __init__(
+        self,
+        coordinator: FrigateDataUpdateCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Construct a FrigateContainerUpdate."""
+        CoordinatorEntity.__init__(self, coordinator)
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID to use for this entity."""
+        return "frigate_container"
```

Comment:
Can't share unique_ids between entities (e.g. multiple Frigate instances on a single HA server would make this break). Suggested (depending on what you think of the naming comment above):

```py
        return get_frigate_entity_unique_id(
            self._config_entry.entry_id, "update", "server"
        )
```

Reply:
Ahhh very interesting, good to know